### PR TITLE
Home screen

### LINKED ISSUE #1: Feature: Home Screen
We need a home screen with a way for players to get into the game.

It should have at least
- [x] the title
- [x] a play button for the main mode

Nice to haves
- [x] a How to Play button that pops up a modal to explain the game
- [ ] a Settings button if we have any settings the user wants to adjust

### PR DESCRIPTION
**Pull request by savwiley**
Created new PR for the home screen to be much smaller and more streamlined.

Branch is largely still a draft, but all that's left is connecting links.

### CODE REVIEW

File: .env
Comment (rankoliang) [inline .env]: What's this for? If it's something you need locally, add `.env` to the `.gitignore` file and delete it from the pr please!

File: src/components/Home/index.js
Comment (rankoliang) [inline src/components/Home/index.js]: There is no rules file in this pr. Also, You the file of the main component should be `src/components/Rules/index.js` and we can import from `../Rules`.

File: src/components/Home/index.js
Comment (rankoliang) [inline src/components/Home/index.js]: Remove this and the useEffect import because rules is not in the pr.

File: src/components/Home/index.js
Comment (rankoliang) [inline src/components/Home/index.js]: Try to use the event handling built directly into react for this. You can put an `onClick` handler on the component. Write a function called `handleClick` in the component and put the callback in there. Then, write
```jsx
<IconLink href="#" onClick={handleClick} />
```
to attach the handler to the component.

File: src/components/Home/index.js
Comment (rankoliang) [inline src/components/Home/index.js]: You can use useRef to get a reference to the object directly without having to add classes or ids.
https://reactjs.org/docs/hooks-reference.html#useref

File: src/components/Home/styled.js
Comment (rankoliang) [inline src/components/Home/styled.js]: Rename this to `IconLink` perhaps?

File: src/components/Home/index.js
Comment (rankoliang) [reply src/components/Home/index.js]: I would also look into conditional rendering instead of changing the style! You can store the state like `rulesShown` and write
```js
{rulesShown && <Rules setShown={setRulesShown} />}
```
The onClick handler on the rules button should toggle the state.

You can use setShown in the Rules component too. I assume you have a close button there, so add an onClick handler to set the state to false to close the modal.

https://reactjs.org/docs/conditional-rendering.html

File: src/components/Home/index.js
Comment (rankoliang) [inline src/components/Home/index.js]: You might want to extract these into their own components, like `<SettingsIcon />` to make it more clear.

File: src/components/Home/styled.js
Comment (rankoliang) [inline src/components/Home/styled.js]: You don't need the & if you're doing it like this. Just
`.title {
}
`
is good enough.

File: src/components/Home/styled.js
Comment (rankoliang) [inline src/components/Home/styled.js]: Use `rems` instead of px for all of the padding and margin

File: src/components/Home/styled.js
Comment (rankoliang) [inline src/components/Home/styled.js]: I would use `em` or `rem` for all of the font-sizes and make sure the elements work relative to each other. If we want to add scaling, I think it would be better to do it globally. What do you think?

File: src/components/Home/styled.js
Comment (rankoliang) [reply src/components/Home/styled.js]: A better way would be to make it into its own `Title` component and extend the styles with `styled(AppHeader)`.
https://styled-components.com/docs/basics#extending-styles

**Review by rankoliang — CHANGES_REQUESTED** [review]
Looks great so far! I like what you did with it all. I just have some changes I would make. Let me know what you think about everything.

On horizontally narrow screen sizes, the icons aren't center. Also the page isn't centered vertically and there is a lot of empty space underneath.

image

File: src/components/Home/index.js
Comment (savwiley) [reply src/components/Home/index.js]: Done but leaving this here as a reminder to fix it in the Rules PR, too.

File: .env
Comment (savwiley) [reply .env]: Added to `.gitignore`, still needs to be deleted.

File: src/components/Icons/HomeIcon.js
Comment (timjacksonm) [inline src/components/Icons/HomeIcon.js:7]: href="#/"
We can go ahead and add this attribute and location.

File: src/components/Icons/LevelButton.js
Comment (timjacksonm) [inline src/components/Icons/LevelButton.js:7]: href="#/levelselect"
We can go ahead and add this attribute and location.

File: src/components/Home/index.js
Comment (timjacksonm) [inline src/components/Home/index.js]: href="#/levels/beginner/1"
We can go ahead and add this attribute and location.

**Review by timjacksonm — APPROVED** [review]
I think this looks good. Lets go ahead and add the attribute href with location. After that is done I say its ready to merge.

**Comment by savwiley** [comment]
Added href attributes but please don't submit yet. I'm working on the Rules modal and have a problem with closing it (ie. changing the state in `Home/index.js` to `false` from `Rules/index.js` via the close event). I want to make sure that's correct first. Until then, I may need to rework more of `Home/index.js`.

**Comment by savwiley** [comment]
Okay, now it's good for another review.

File: src/components/Icons/HomeIcon.js
Comment (timjacksonm) [reply src/components/Icons/HomeIcon.js:7]: removed my suggestion. The decision on best resolution was resolved via discord.

File: src/components/Icons/LevelButton.js
Comment (timjacksonm) [reply src/components/Icons/LevelButton.js:7]: same as above. Please update too
href="/levels"
I also made a mistake here of adding the hash symbol #.
Also made the error of not following what the url should be. Should be levels not levelselect

**Review by timjacksonm — CHANGES_REQUESTED** [review]
A few updates due to my mistake! Sorry! This should work now!

File: src/components/Home/index.js
Comment (timjacksonm) [inline src/components/Home/index.js]: to="/levels"
just levels now would be correct. My mistake on telling you otherwise

File: src/components/Icons/HomeIcon.js
Comment (timjacksonm) [reply src/components/Icons/HomeIcon.js:7]: This is perfect! :+1:

File: src/components/Icons/LevelButton.js
Comment (timjacksonm) [reply src/components/Icons/LevelButton.js:7]: to="/levels"
just levels now would be correct. My mistake on telling you otherwise.

**Review by timjacksonm — CHANGES_REQUESTED** [review]
Just reviewing your most recent commit. Thanks for updating this again.

File: src/components/Home/index.js
Comment (savwiley) [reply src/components/Home/index.js]: Fixed these links in the merged Rules PR.

Comment (savwiley) [reply src/components/Home/index.js]: Fixed

File: src/components/Icons/styled.js
Comment (rankoliang) [inline src/components/Icons/styled.js]: This should be a `button` since it pops up a modal. You can also extend the IconLink styles and add an `as="button"`, so something like
```js
export const RulesButton = styled(IconLink).attrs({
  as: "button"
}`
  // css
`
```
could work.
https://stackoverflow.com/questions/55776961/styled-components-extend-styles-and-change-element-type

File: src/components/Icons/styled.js
Comment (rankoliang) [inline src/components/Icons/styled.js]: The scrollbar pops up for me when the hover transformation occurs. You can fix this by making the `transform-origin` the bottom right, or setting `overflow: hidden` on the App component.

**Review by rankoliang — APPROVED** [review]
One slight issue with the RulesLink. Other than that, LGTM! You can merge it when you're done fixing that.

**Pull request merged**
